lib/livebook/zta/livebook_teams.ex

```diff
@@ -0,0 +1,147 @@
+defmodule Livebook.ZTA.LivebookTeams do
+  use GenServer
+  use LivebookWeb, :verified_routes
+
+  require Logger
+
+  alias Livebook.Teams
+
+  import Plug.Conn
+  import Phoenix.Controller
+
+  @behaviour Livebook.ZTA
+
+  @impl true
+  def child_spec(opts) do
+    %{id: __MODULE__, start: {__MODULE__, :start_link, [opts]}}
+  end
+
+  def start_link(opts) do
+    name = Keyword.fetch!(opts, :name)
+
+    if id = Livebook.Config.teams_auth_hub_id() do
+      team = Livebook.Hubs.fetch_hub!(id)
+      Livebook.ZTA.put(name, team)
+
+      :ignore
+    end
+  end
+
+  @impl true
+  def init(arg) do
+    {:ok, arg}
+  end
+
+  @impl true
+  def authenticate(name, conn, _opts) do
+    team = Livebook.ZTA.get(name)
+    auth = Livebook.Config.teams_auth()
```

Comment:
We don't need to read this or pass it around, we will only work on online for now.